File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
    if (!input || !input.trim() || 0 === input.length) {
        return 'Fine. Be that way!';
    }

    // the regex checks if there's at least a letter in the string, either uppercase or not
    if (/[a-zA-Z]/.test(input) && (input === input.toUpperCase())) {
            return 'Whoa, chill out!';
    }

    if (input[input.length - 1] == '?') {
        return 'Sure.';
    }

    // in all the other cases:
    return 'Whatever.';
};

module.exports = Bob;

File: python/bob/bob.py
import re

def hey(what):
	what = what.strip()
	
	if not(what.strip()) or 0 == len(what):
		return "Fine. Be that way!"
	if re.search('[A-Z]', what) and what == what.upper():
		return "Whoa, chill out!"
	if "?" == what[len(what) - 1]:
		return "Sure."
	
	return "Whatever."
